compiler/src/iree/compiler/Codegen/Dialect/UKernelOps.cpp

```diff
@@ -99,15 +99,13 @@ static LogicalResult getCallOpType(MLIRContext *context,
         callOperandTypes.push_back(MemRefType::get(
             ArrayRef<int64_t>{}, memrefType.getElementType(),
             MemRefLayoutAttrInterface{}, memrefType.getMemorySpace()));
-        if (memrefType.getRank() == 0) return success();
-
-        auto indexType = IndexType::get(context);
         // Offset
+        auto indexType = IndexType::get(context);
         callOperandTypes.push_back(indexType);
-
-        if (bufferPassingStyle ==
-            BufferPassingStyle::kBufferAndOffsetAndNm1Strides) {
-          // Strides.
+        // Strides.
+        if (memrefType.getRank() >= 1 &&
+            bufferPassingStyle ==
+                BufferPassingStyle::kBufferAndOffsetAndNm1Strides) {
```

Comment:
Interesting. I dont think I have seen `BufferPassingStyle` but seems like a good idea.... Is it possible to change `BufferPassingStyle::kBufferAndOffsetAndNm1Strides` -> `BufferPassingStyle::kBufferOffsetSizesAndStrides` , i.e. state the buffer is lowered to 1 offsets , `n` sizes and `n` strides. If this breks the current lowering paths, you can defer that (or punt it to me) to do after you move all the existing micro kernel ops to the form you want

Reply:
`BufferPassingStyle` is introduced in #13053, the previous PR in this queue.

The currently explicit enum value names are intentional to call out such discrepancy between current state of things and intention.  How about we first converge on what the code does while retaining explicit descriptive enum names, then once that's stabilized we rename (maybe).

Currently 0 sizes are passed and (n-1) strides are passed.  The main friction against changing this immediately is that currently the `matmul` ukernel goes through this generic path, and it does want exactly that.  The PR queue I'm building will involve migrating other ukernels to UKernelOps, then retiring the matmul ukernel, which will remove this friction.